### PR TITLE
Python: Fix bad join in `import_star_read`

### PR DESCRIPTION
**Pull request by tausbn**
Makes this

```
(21s) Tuple counts for DataFlowPublic::import_star_read#ff/2@fcd5e6nr after 8.5s:
9743      ~6%     {3} r1 = SCAN num#DataFlowPublic::TModuleVariableNode#fff OUTPUT In.1, In.0, In.2 'result'
9743      ~1%     {3} r2 = JOIN r1 WITH Variables::Variable::getId_dispred#ff ON FIRST 1 OUTPUT Rhs.1, Lhs.1, Lhs.2 'result'
390808917 ~3%     {3} r3 = JOIN r2 WITH Flow::NameNode::getId_dispred#ff_10#join_rhs ON FIRST 1 OUTPUT Rhs.1, Lhs.1, Lhs.2 'result'
307       ~0%     {2} r4 = JOIN r3 WITH ImportStar::ImportStar::importStarResolvesTo#ff ON FIRST 2 OUTPUT Lhs.0, Lhs.2 'result'
307       ~0%     {2} r5 = JOIN r4 WITH num#DataFlowPublic::TCfgNode#ff ON FIRST 1 OUTPUT Rhs.1 'n', Lhs.1 'result'
                  return r5
```

become this

```
(17s) Tuple counts for DataFlowPublic::resolved_import_star_module#fff/3@f5e84aic after 0ms:
307 ~0%     {3} r1 = JOIN ImportStar::ImportStar::importStarResolvesTo#ff WITH num#DataFlowPublic::TCfgNode#ff ON FIRST 1 OUTPUT Lhs.0, Lhs.1 'm', Rhs.1 'n'
307 ~0%     {3} r2 = JOIN r1 WITH Flow::NameNode::getId_dispred#ff ON FIRST 1 OUTPUT Lhs.1 'm', Rhs.1 'name', Lhs.2 'n'
            return r2
(17s) Registering DataFlowPublic::resolved_import_star_module#fff/3@f5e84aic +  with content f29281ig38r98icro4ege09mrva
(17s)  >>> Created relation DataFlowPublic::resolved_import_star_module#fff/3@f5e84aic with 307 rows.
(17s) Starting to evaluate predicate DataFlowPublic::import_star_read#ff/2@57b0c06e
(17s) Tuple counts for DataFlowPublic::import_star_read#ff/2@57b0c06e after 2ms:
9743 ~0%     {3} r1 = SCAN num#DataFlowPublic::TModuleVariableNode#fff OUTPUT In.1, In.0, In.2 'result'
9743 ~0%     {3} r2 = JOIN r1 WITH Variables::Variable::getId_dispred#ff ON FIRST 1 OUTPUT Lhs.1, Rhs.1, Lhs.2 'result'
307  ~0%     {2} r3 = JOIN r2 WITH DataFlowPublic::resolved_import_star_module#fff ON FIRST 2 OUTPUT Rhs.2 'n', Lhs.2 'result'
             return r3
```

--- 

Needs performance evaluation. No change note required.

### CODE REVIEW

File: python/ql/lib/semmle/python/pointsto/PointsTo.qll [hunk 1]
```diff
@@ -2125,7 +2125,7 @@ module Conditionals {
 /** INTERNAL: Do not use. */
 predicate declaredAttributeVar(PythonClassObjectInternal cls, string name, EssaVariable var) {
   name = var.getName() and
-  var.getAUse() = cls.getScope().getANormalExit()
+  pragma[only_bind_into](pragma[only_bind_into](var).getAUse()) = cls.getScope().getANormalExit()
```
Comment (yoff) [inline python/ql/lib/semmle/python/pointsto/PointsTo.qll:2128]: Interesting construction :-)

**Review by yoff — APPROVED** [review]
Solutions look reasonable, and numbers look very persuasive :-)

**Comment by yoff** [comment]
Do you want to wait for a DCA run to see the full effect?

**Comment by tausbn** [comment]
> Do you want to wait for a DCA run to see the full effect?

Yeah. I don't expect these changes to have bad consequences, but there's no reason to chance it.

File: python/ql/lib/semmle/python/pointsto/PointsTo.qll (hunk 1 above)
Comment (tausbn) [reply python/ql/lib/semmle/python/pointsto/PointsTo.qll:2128]: Yeah, it's not the most elegant construction, but a single `pragma` just didn't do the trick.

**Comment by tausbn** [comment]
The performance data is in, and I didn't find any hidden surprises.

**Comment by tausbn** [comment]
This had gotten a bit stale, so I rebased and restarted the performance tests.

**Review by yoff — APPROVED** [review]
New evaluation looks fine, so ready to merge this.

**Pull request merged**